feat: Obsidian theme to use current Omarchy theme

darcy opened the pull request:
## Overview

Scripts to install, update, and remove Obsidian theme based on Omarchy's current theme.

## New Scripts
```bash
omarchy-obsidian-theme-install
```
* TUI with `gum` for user to navigate and select to Obsidian Vault.
* Registers the vault location in `.local/state/omarchy/obsidian-vaults
* Can register multiple vaults
* Installs `manifest.json` and an empty `theme.css` in `.obsidian/themes/Omarchy/`
* Triggers new script `omarchy-obsidian-theme-update`
* In Obsidian, a new 'Omarchy' theme is installed and available to use

```bash
omarchy-obsidian-theme-update
```
* Looks for `obsidian.css` in current theme, if present, replaces Obsidian theme with this content
* If not present, reads multiple theme files to infer the values for an effective Obsidian theme. (see video, these turn out really good defaults)
* If theme is selected in Obsidian, then the theme is immediately updated with changes.
 
```bash
omarchy-obsidian-theme-remove
```
* TUI to select the registered vault to uninstall

## Omarchy Hooks

* Menu triggers: 
  * Install > Style > Obsidian Theme (wasn't a great spot to stick this, this felt like best place)
  * Remove > Obsidian Theme
* `omarchy-theme-set` this will immediately trigger `omarchy-obsidian-theme-update`
* `omarchy-theme-update` this will immediately trigger `omarchy-obsidian-theme-update`

## Demo
https://github.com/user-attachments/assets/e7efdc45-7d98-4fd3-8da0-b29ae3a12cef

dhh commented:
I do love this idea. I want it for my setup. But need to review the implementation in detail, so will need to wait until next release.

dhh commented:
I'm very much digging this, but I'm finding the vault installation stuff a bit cumbersome. What I'd love to see is that we simply add an Omarchy theme for every single vault we find/see. That theme is updated via omarchy-theme-set, and if that theme is set in Obsidian, then you get the live switching. What do you think?

darcy commented:
@dhh This is updated with the following changes:
* Removed install/remove scripts and flow
* First `omarchy-theme-set-obsidian` run will look for vaults in Documents and Dropbox folders and add them to `/.local/state/omarchy/obsidian-vaults`
  * Point of this is such that we assume a couple locations to look for vaults but allow people to add vaults to the file for custom locations. 
  * Subsequent omarchy-theme-set invocations aren't impacted by find/search on large file systems, which was a concern of mine.
  * Note there is no documentation about this registry at the moment for people to know they can manually put custom locations of their vaults.
* Each `omarchy-theme-set` invocation will install the themes on all found/registered vaults if missing and update the live theme
* Added an option of `omarchy-theme-set-obsidian --reset` to wipe out themes in registered vaults and remove the registry file. This would be the option to re-run to the automatic vault registration.
* Added migration to trigger install immediately in Obsidian, note that you still need to pick Omarchy theme.

I looked into an Omarchy community theme for Obsidian, that looked to be too messy an approach for a few reasons - starting with it needs to be in a separate repo and would likely require it to be a plugin rather than a theme for the wiring up to work smoothest.
